### ISSUE TITLE
feat: support new Error.cause property

### ISSUE THREAD

**Issue by petermetz** [ShogunPanda/http-errors-enhanced#6]
It would be great if this library could pass the `cause` property down to the underlying (sub-classed) built-in `Error` constructor so that something like this works:

```typescript
const cause = new Error("Something went wrong and this is the root cause");
throw new InternalServerError("An internal error occurred.", { cause });
```

In comparison if you write this code it will print the screenshot below (that's the expected behavior that would be great to get from this library as well):

```typescript
const cause = new Error("Something went wrong and this is the root cause");
throw new Error("An internal error occurred.", { cause });
```

![image](https://github.com/user-attachments/assets/ff7e4227-fa60-4a59-8c4e-ef1fa2740c22)

I believe all it needs is for the `properties` to be passed in at this line in base.ts:

```typescript
    // Assign basic properties
    super(message as string)

```

https://github.com/ShogunPanda/http-errors-enhanced/blob/main/src/base.ts#L41

![image](https://github.com/user-attachments/assets/1df9fbb1-5cfe-40c7-894a-9da9d29d49f8)

https://github.com/tc39/proposal-error-cause

**Comment by ShogunPanda** [comment]
Hello! I included this in 3.0.0 release!

**Comment by petermetz** [comment]
@ShogunPanda Thank you so much!
